### DISCUSSION TITLE
How to use :update?

### DISCUSSION THREAD

**Discussion by Madd0g** [cozodb/cozo#142] [Q&A]
I'm wondering how to use :update and what's possible to update? only value columns?

for example, in #140 - there's this relation mentioned:

```
:create beliefs {
    belief_id: Uuid,
    character_id: Uuid,
    belief: String,
    last_accessed_at: Validity default [floor(now()), true],
    =>
    details: String default "",
    parent_belief_id: Uuid? default null,
    valence: Float default 0,
    aspects: [(String, Float, String, String)] default [],
    belief_embedding: <F32; 768>,
    details_embedding: <F32; 768>,
}
```

and I see `last_accessed_at` there as part of the keys - does that mean that keys can be updated? after all why have validity/last_anything if you can't update it?

I've only seen validity used in examples as a one of the values, not as one of the keys - does that make any difference regarding what it can be used for?

How to do an update on a relation like the above? Actually same question for `:replace`, which also doesn't have an example.

Thanks

**Comment by Madd0g** [comment]
never mind, figured it out, it's the same mechanics as `:create`, I hadn't realized it's that easy.

**Reply by jamesfebin** [reply]
Can you please give an example query, struggling with this.

**Reply by jamesfebin** [reply]
I figured it out; here's a basic CRUD example for people who are still struggling.

Creating stored relation
```
:create person{
id : Int,
=>
name : String,
Age: Int
}
```

Inserting a row 
```
?[id, name, Age] <- [[1, "Alice", 30]]
:put person {id, name, Age}
```

Fetching a row 
```
?[id, name, Age] := *person[id, name, Age], Age>20
```

Updating a row
```
?[id, name] := id = 1, *person{id}, name = "John"
:update person {id => name}
```

Deleting a row
```
?[id] := id = 1, *person{id}
:rm person {id}
```

**Comment by vrn21-arcanist** [comment]
can you look at <a href="https://github.com/cozodb/cozo/issues/271#issue-2388116689">here </a>
